jupyter_client/multikernelmanager.py

```diff
@@ -260,10 +260,11 @@ async def _async_shutdown_kernel(
         if self._using_pending_kernels() and kernel_id in self._pending_kernels:
             raise RuntimeError("Kernel is in a pending state. Cannot shutdown.")
         # If the kernel is still starting, wait for it to be ready.
-        elif kernel_id in self._starting_kernels:
-            kernel = self._starting_kernels[kernel_id]
+        elif kernel_id in self._pending_kernels:
```

Comment:
Since `_starting_kernels` is a shim for `_pending_kernels` defined here:

https://github.com/jupyter/jupyter_client/blob/263173095ceab87a572937f21f843886ffe356f1/jupyter_client/multikernelmanager.py#L102-L105

We might as well use `_pending_kernels` directly.